casts/dateTimePicker/src/UIDateTimePicker/Calendar/useCalendar.ts

```diff
@@ -100,13 +97,20 @@ export function useDaysCalendar() {
     } = useDateTimeState();
     const date = React.useMemo(() => selectedDate ?? dayjs(), [selectedDate]);
 
-    const dayNamesShort = React.useMemo(
-        () => Object.values(uiLocalized.DateTimePicker.dayNamesShort),
-        [],
-    );
+    // eslint-disable-next-line radix
+    const firstDayOfWeek: number = React.useMemo(() => uiLocalized.getFirstDayOfWeek(), []);
+
+    const dayNamesShort = React.useMemo(() => {
+        const arr = Object.values(uiLocalized.DateTimePicker.dayNamesShort);
+        if (arr && firstDayOfWeek) {
+            const arr2 = arr.shift();
+            arr2 && arr.push(arr2);
+        }
```

Comment:
What if in some mystic country the firstDatOfTheWeek is Wednesday? =)
I believe that you need to apply the "shift" as much time as it's required